draft-irtf-cfrg-bbs-signatures.md

```diff
@@ -265,6 +268,23 @@ Throughout the operations of this signature scheme, each message that is signed
 
 Aside from the message generators, the scheme uses two additional generators: `Q_1` and `Q_2`. The first (`Q_1`), is used for the blinding value (`s`) of the signature. The second generator (`Q_2`), is used to sign the signature's domain, which binds both the signature and generated proofs to a specific context and cryptographically protects any potential application-specific information (for example, messages that must always be disclosed etc.).
 
+### Serializing to octet strings
+
+When serializing different types of elements (i.e., Points, Scalars etc.) to get their octet representation, each element will be serialized with a specific operation, depending on its type. More concretely,
+
+- Points in `G*` will be serialized using the `point_to_octets_g*` implementation for a particular ciphersuite.
+- Non-negative integers will be serialized using `I2OSP` with an output length of 8 bytes.
+- Scalars will be serialized using `I2OSP` with a constant output length defined by a particular ciphersuite.
+- ASCII strings will be transformed into octet strings using UTF-8 encoding.
+
+Variable length octet strings, will be prepended with a value representing the length of their binary representation in the form of the number of bytes. This length must be encoded to octets using I2OSP with output length of 8 bytes. The combined value (encoded value + length prefix) binary representation is then encoded as a single octet string. For example, the string `0x14d` will be encoded as `0x0000000000000002014d`. If the length of the octet string is larger than 2^64 - 1, the octet string must be rejected. Similarly, ASCII strings, after encoded to octets (using utf8), will also be appended with the length of their octet-string representation.
+
+Constant length octet strings (e.g., the ciphersuite id octet representation), will be encoded directly.
+
+Optional input/parameters to operations that feature in a call to hash\_to\_scalar, that are not supplied to the operation should default to an empty octet string. For example, if X is an optional input/parameter that is not supplied, whilst A and B are required, then the procedural step of `hash(A || X || B)` MUST be evaluated to `hash(A || "" || B)`.
```

Comment:
I think we should just be explicit about what the defaults are for each variable at the point of use

Reply:
Agreed. Removed the paragraph on optional inputs. For constant length octet strings will revisit after #232 is resolved.